### DISCUSSION TITLE
Could you explain some thing in Test_Main

### DISCUSSION THREAD

**Discussion by zebox** [umputun/reproxy#136] [General]
I research you code for borrow some approach in my project and i don't understand some way in `Test_Main`.
If some flags will has error when `Parse` method doing, whole test exit `os.Exit(2)` and [test](https://github.com/umputun/reproxy/blob/f9eb8ac0128d3de14b627c4840ab84d9641ec216/app/main_test.go#L43:#L46) doesn't complete successfully (without error or any callback).

```go 
if _, err := p.Parse(); err != nil {
	if err.(*flags.Error).Type != flags.ErrHelp {
		log.Printf("[ERROR] cli error: %v", err)
	}
       os.Exit(2)
}
```
Or is this test doesn't imply for testing with incorrect flags?
I'm thinking this isn't graceful way for test, or I misunderstand something.

**Comment by umputun** [comment]
I'm not sure I fully understand the issue you are describing, but generally I don't want to test external packages/dependencies, so I don't bother to isolate flag parsing and make it truly testable. If, for some reason, you do - just make the parseFlag func and test it with a usual unit test. Test_Main is not a unit test, but rather integration test making sure the whole app started properly and responds reasonably

**Reply by zebox** [reply]
That's right! It's the integration test and this main reason of my misunderstood. I mixed up it with unit test and try testing starting server with various flags 🤦, but it's inccorect in this case. 
I made different unit test for testing options flag.

Thank you for you reply!
